src/textual/widgets/_text_area.py

```diff
@@ -1032,6 +1017,21 @@ def render_line(self, widget_y: int) -> Strip:
                         else:
                             line.stylize(selection_style, end=line_character_count)
 
+        highlights = self._highlights
```

Comment:
Just reordering the style sandwich ingredients here, no actual change to the code that is inside the diff.